### PR TITLE
added a very famous question of linked list to find the middle elemen…

### PR DESCRIPTION
**Pull request by sunilkumarmaurya786693**
### This is a famous interview question to find the middle element of linked list in single traversal of linked list.
Given below the question link and my solution approach.
**question link ----->
https://leetcode.com/problems/middle-of-the-linked-list/**

**Solution--->**
Each time, slow go 1 steps while fast go 2 steps.
When fast arrives at the end, slow will arrive right in the middle.

    ```
ListNode* middleNode(ListNode* head) {
        ListNode *slow = head, *fast = head;
        while (fast && fast->next)
            slow = slow->next, fast = fast->next->next;
        return slow;
    }
```

### CODE REVIEW

File: leetcode/linkedlist/i.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+5
+1 2 3 4 5
```
Comment (sunilkumarmaurya786693) [inline leetcode/linkedlist/i.txt:2]: test case should also cover corner case.

File: leetcode/linkedlist/middle.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
+#include <bits/stdc++.h>
+#define pii pair<int, int>
+#define inf_neg -100000000000000ll
+using namespace std;
+typedef long long int ll;
+ll m = 1000000007;
+
+// 5
```
Comment (sunilkumarmaurya786693) [inline leetcode/linkedlist/middle.cpp:8]: delete comment please.
